### PR TITLE
Feature 21 task list and create

### LINKED ISSUE #21: Tasks list and create
Tasks list view and create a new task functionality

### PR DESCRIPTION
**Pull request by Daniilmass**
Resolves #21

### CODE REVIEW

File: src/routes/projects_.$projectId.schedule_.$milestoneId.tasks.tsx [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+  const findProjectMilestoneQuery = useFindProjectMilestoneQuery(projectId, milestoneId);
+  const milestone = findProjectMilestoneQuery.data;
+  const listMilestoneTasksQuery = useListMilestoneTasksQuery({ projectId, milestoneId });
+  const tasks = listMilestoneTasksQuery.data;
+
+  const [open, setOpen] = React.useState(false);
+  const [task, setTask] = React.useState<null | Task>(null);
+
+  /**
+   * Handles task select
```
Comment (Jdallos) [inline src/routes/projects_.$projectId.schedule_.$milestoneId.tasks.tsx:51]: Missing @param

File: src/routes/projects_.$projectId.schedule_.$milestoneId.tasks.tsx [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+      const formattedEndDate = endDate.toFormat("dd.MM.yyyy");
+
+      return (
+        <TableRow key={task.id}>
+          <TableCell style={{ overflow: "hidden" }}>
+            <div style={{ display: "flex", alignItems: "center", gap: 8 }}>
+              <Avatar sx={{ backgroundColor: "#0079BF" }}>
+                <FlagOutlinedIcon fontSize="large" sx={{ color: "#fff" }} />
+              </Avatar>
+              {/* TODO: Handle overflowing name with maxWidth could be improved */}
```
Comment (Jdallos) [inline src/routes/projects_.$projectId.schedule_.$milestoneId.tasks.tsx:94]: Again if you think this is fine, feel free to remove the comment

**Review by Jdallos — APPROVED** [review]

File: src/components/tasks/task-dialog.tsx [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+                <TableCell>{t("newMilestoneTaskDialog.taskConnectionsTable.type")}</TableCell>
+                <TableCell>{t("newMilestoneTaskDialog.taskConnectionsTable.task")}</TableCell>
+                <TableCell>{t("newMilestoneTaskDialog.taskConnectionsTable.status")}</TableCell>
+              </TableRow>
+            </TableHead>
+            <TableBody>
+              <TableRow>
+                <TableCell>Type 1</TableCell>
+                <TableCell>Task name</TableCell>
+                <TableCell>Status</TableCell>
```
[the remark this answers is not in the document]
Comment (Daniilmass) [reply src/components/tasks/task-dialog.tsx:306]: Yes, those are placeholders

File: src/routes/projects_.$projectId.schedule.tsx [hunk 4]
```diff
@@ -63,20 +63,25 @@ function ScheduleIndexRoute() {
...
+            <Link
+              to={`/projects/${projectId}/schedule/${milestone.id}/tasks` as string}
+              style={{ textDecoration: "none", color: "#000" }}
+            >
+              <div style={{ display: "flex", alignItems: "center", gap: 8, cursor: "pointer" }}>
+                <Avatar sx={{ backgroundColor: "#0079BF" }}>
+                  <FlagOutlinedIcon fontSize="large" sx={{ color: "#fff" }} />
+                </Avatar>
+                {/* TODO: Handle overflowing name with maxWidth could be improved */}
+                <Box sx={{ margin: "0 1rem", maxWidth: 300 }}>
```
[the remark this answers is not in the document]
Comment (Daniilmass) [reply src/routes/projects_.$projectId.schedule.tsx:75]: Let's keep the comment for now :)

File: src/routes/projects_.$projectId.schedule_.$milestoneId.tasks.tsx (hunk 2 above)
Comment (Daniilmass) [reply src/routes/projects_.$projectId.schedule_.$milestoneId.tasks.tsx:94]: Let's keep it for now, maybe we can verify and remove it in our future PRs

**Review by Jdallos — APPROVED** [review]

**Pull request merged**